src/gnutls.c

```diff
@@ -26,7 +26,7 @@ along with GNU Emacs.  If not, see <https://www.gnu.org/licenses/>.  */
 #include "coding.h"
```

Comment:
gnutls test are faling. though all passes locally.

Reply:
I can reproduce that in a Trusty docker. Looking at it now.

Reply:
GNUTLS version numbering is the culprit.This commit https://github.com/emacs-mirror/emacs/commit/583995c62dd424775dda33d5134ce04bee2ae685 is what broke things.

The build checks for gnutls but does not enforce version >= 3. Trusty has 2.6 and that is what our docker build is running. So the elisp tests assume that GNUTLS works because `gnutls-available-p` is true but some of the functions are only in version 3 and there is no check for the supported version in the elisp. This is a bug upstream. Either the configure should fail or it should detect the GNUTLS version and let the user know their version is no longer supported. Barring this, the elisp code should have the ability to detect the version supported or at the very least check if the v3 functions are defined before using them.